cmd/zoekt-sourcegraph-indexserver/index.go

```diff
@@ -156,30 +192,88 @@ func gitIndex(o *indexArgs, runCmd func(*exec.Cmd) error) error {
 		return err
 	}
 
-	fetchStart := time.Now()
+	var fetchCommits = func(commits []string) error {
+		fetchArgs := []string{"-C", gitDir, "-c", "protocol.version=2", "fetch", "--depth=1", o.CloneURL}
+		fetchArgs = append(fetchArgs, commits...)
+
+		cmd = exec.CommandContext(ctx, "git", fetchArgs...)
+		cmd.Stdin = &bytes.Buffer{}
+
+		return runCmd(cmd)
+	}
+
+	successfullyFetchedCommitsCount := 0
+
+	var fetchDuration time.Duration
+	allFetchesSucceeded := true
+
+	defer func() {
+		success := strconv.FormatBool(allFetchesSucceeded)
+		name := repoNameForMetric(o.Name)
+		metricFetchDuration.WithLabelValues(success, name).Observe(fetchDuration.Seconds())
+	}()
 
 	// We shallow fetch each commit specified in zoekt.Branches. This requires
 	// the server to have configured both uploadpack.allowAnySHA1InWant and
 	// uploadpack.allowFilter. (See gitservice.go in the Sourcegraph repository)
```

Comment:
nit this comment belongs above the moved fetchArgs. It is explaining the --depth=1 arg